packages/ember/tests/component_context_test.js

```diff
@@ -0,0 +1,145 @@
+    });
+  }
+  ['@test Components without a block should have the proper content when a template is provided'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{my-component}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      text: 'inner'
+    }), template: '{{text}}' });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'inner', 'The component is composed correctly');
+    });
+  }
+
+  ['@test Components without a block should have the proper content'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{my-component}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      didInsertElement() {
+        this.$().html('Some text inserted by jQuery');
+      }
+    }) });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'Some text inserted by jQuery', 'The component is composed correctly');
+    });
+  }
+
+  ['@test properties of a component without a template should not collide with internal structures [DEPRECATED]'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{my-component data=foo}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer',
+      'foo': 'Some text inserted by jQuery'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      didInsertElement() {
+        this.$().html(this.get('data'));
+      }
+    }) });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'Some text inserted by jQuery', 'The component is composed correctly');
+    });
+  }
+
+  ['@test attrs property of a component without a template should not collide with internal structures'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{my-component attrs=foo}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer',
+      'foo': 'Some text inserted by jQuery'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      didInsertElement() {
+        // FIXME: I'm unsure if this is even the right way to access attrs
+        this.$().html(this.get('attrs.attrs.value'));
+      }
+    }) });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'Some text inserted by jQuery', 'The component is composed correctly');
+    });
+  }
+
+  ['@test Components trigger actions in the parents context when called from within a block'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}<a href=\'#\' id=\'fizzbuzz\' {{action \'fizzbuzz\'}}>Fizzbuzz</a>{{/my-component}}</div>');
```

Comment:
Wrapping the templates in backticks would avoid all the quote escaping here. In addition backticks would permit multiple lines:

```js
this.addTemplate('application', `
  <div id="wrapper">
    {{#my-component}}
      <a href="#" id="fizzbuzz" {{action 'fizzbuzz'}}>
        Fizzbuzz
      </a>
    {{/my-component}}
  </div>
`);
```

Much easier to grok.